src/cargo/ops/cargo_add/mod.rs

```diff
@@ -469,6 +477,26 @@ fn check_invalid_ws_keys(toml_key: &str, arg: &DepOp) -> CargoResult<()> {
     Ok(())
 }
 
+/// When the `--optional` option is added using `cargo add`, we need to
+/// check the current rust-version. As the `dep:` syntax is only avaliable
+/// starting with Rust 1.60.0
+///
+/// `true` means that the rust-version is None or the rust-version is higher
+/// than the version needed.
+///
+/// Note: Previous versions can only use the implicit feature name.
+fn check_rust_version_for_optional_dependency(
+    rust_version: Option<&RustVersion>,
+) -> CargoResult<bool> {
+    match rust_version {
+        Some(version) => {
+            let syntax_support_version = String::from("1.60.0");
+            Ok(version.to_string().cmp(&syntax_support_version).is_ge())
```

Comment:
string comparisons aren't accurate.  I'd recommend directly constructing a `RustVersion` and using that for the comparison

Reply:
thanks and solved.